tests/e2e/main.go

```diff
@@ -287,226 +283,121 @@ func getTestCases(selectedPredefinedTests, selectedTestFiles TestSet) (tests []t
-	case ForkConsumerChainAction:
-		tr.forkConsumerChain(action, verbose)
-	case UpdateLightClientAction:
-		tr.updateLightClient(action, verbose)
-	case StartConsumerEvidenceDetectorAction:
-		tr.startConsumerEvidenceDetector(action, verbose)
-	case SubmitChangeRewardDenomsProposalAction:
-		tr.submitChangeRewardDenomsProposal(action, verbose)
-	default:
-		log.Fatalf("unknown action in testRun %s: %#v", tr.name, action)
+	if len(consumerVersions) == 0 {
+		consumerVersions = append(consumerVersions, "")
 	}
-
-	modelState := step.State
-	actualState := tr.getState(step.State)
-
-	// Check state
-	if !reflect.DeepEqual(actualState, modelState) {
-		fmt.Printf("=============== %s FAILED ===============\n", tr.name)
-		fmt.Println("FAILED action", reflect.TypeOf(step.Action).Name())
-		pretty.Print("actual state", actualState)
-		pretty.Print("model state", modelState)
-		log.Fatal(`actual state (-) not equal to model state (+): ` + pretty.Compare(actualState, modelState))
+	if len(providerVersions) == 0 {
+		providerVersions = append(providerVersions, "")
 	}
-}
 
-// executeSteps sequentially runs steps.
-func (tr *TestConfig) executeSteps(steps []Step) {
-	fmt.Printf("=============== started %s tests ===============\n", tr.name)
-
-	start := time.Now()
-	for i, step := range steps {
-		// print something the show the test is alive
-		fmt.Printf("running %s: step %d == %s \n",
-			tr.name, i+1, reflect.TypeOf(step.Action).Name())
-		tr.runStep(step, *verbose)
+	for _, provider := range providerVersions {
+		for _, consumer := range consumerVersions {
+			targetCfg.consumerVersion = consumer
+			targetCfg.providerVersion = provider
+			target := DockerContainer{targetConfig: targetCfg}
+			err := target.Build()
+			if err != nil {
+				deleteTargets(targets)
+				return nil, err
+			}
+			targets = append(targets, &target)
+		}
 	}
-
-	fmt.Printf("=============== finished %s tests in %v ===============\n", tr.name, time.Since(start))
+	return targets, nil
 }
 
-func (tr *TestConfig) startDocker() {
-	fmt.Printf("=============== building %s testRun ===============\n", tr.name)
-	localSdk := tr.localSdkPath
-	if localSdk == "" {
-		localSdk = "default"
-	}
-	useGaia := "false"
-	gaiaTag := ""
-	if tr.useGaia {
-		useGaia = "true"
-		if tr.gaiaTag != "" {
-			majVersion, err := strconv.Atoi(tr.gaiaTag[1:strings.Index(tr.gaiaTag, ".")])
-			if err != nil {
-				panic(fmt.Sprintf("invalid gaia version %s", tr.gaiaTag))
-			}
-			if majVersion < 9 {
-				panic(fmt.Sprintf("gaia version %s is not supported - supporting only v9.x.x and newer", tr.gaiaTag))
+func createTestRunners(targets []ExecutionTarget, testCases []testStepsWithConfig) []TestRunner {
+	runners := []TestRunner{}
+	for _, target := range targets {
+		for _, tc := range testCases {
+			tr := TestRunner{
+				config:  tc.testRun,
+				steps:   tc.steps,
+				target:  target,
+				verbose: *verbose,
 			}
-			gaiaTag = tr.gaiaTag
+			//TODO: refactor this target specific setting
```

Comment:
this TODO will be addressed in a different PR, right? we can leave it, I'm just a bit concerned about it never getting addressed